### ISSUE TITLE
Result.ok() gone in 5.x

### ISSUE THREAD

**Issue by krivachy** [true-myth/true-myth#261]
We're upgrading from 4.1.1 to 5.1.0 and one large change we needed to do in our codebase was to replace `Result.ok()` with `Result.ok(Unit)` or else we were getting:

```
Error: Tried to construct `Ok` with `null`. Maybe you want `Maybe.Nothing`?
```

Two things I'd like to ask:

1. Why is this a runtime check instead of a compile time check? The `.d.ts` files contain `export declare function ok<T, E>(): Result<Unit, E>;` so all our code typechecked, but it blows up in runtime. If this is not a function call users of the library should not make then I think ideally it shouldn't typecheck.
2. Can we get this convenience method back? One of the nice things about true-myth is that `Result`, `Maybe`, and `Unit` types are nicely connected to each other and this was one of the nicenesses. On using `Maybe` instead of `Result`: I think `Maybe<ErrorType>` is not as clear as `Result<Unit, ErrorType>` in terms of what the code is doing.

Thanks!

**Comment by chriskrycho** [comment]
Huh, whoops—this is a bug, not intentional. I think this probably was an accidental effect of the work I did to collapse the types all into a single class, but *this* one wasn't an intentional breaking change for sure. The convenience constructors were meant to keep working exactly as they were before. I'll take a look and try to get out a 5.1.1 with a fix sometime in the next few evenings!

**Comment by chriskrycho** [comment]
Ah, yes, that's exactly what happened. Historically we had a bit of a weird misalignment around the constructors: you could call `Result.ok()`, which was actually a sneakily-exported standalone function which constructed an `Ok` class instance, or you could call `new Ok()`. If you called `Result.ok()`, you got back `Result<Unit, unknown>`. If you called `new Ok()`, you got the error you hit here. When I folded the `Ok` and `Err` classes into a single `Result` class, I accidentally brought along the error instead of the convenience function behavior~!

**Comment by chriskrycho** [comment]
Fix released in [v5.1.1](https://github.com/true-myth/true-myth/releases/tag/v5.1.1). Thanks again for the report!

**Comment by krivachy** [comment]
Thanks for the speedy fix @chriskrycho! We love to use true-myth 🔮 in our codebase ❤️
